### ISSUE TITLE
I'm getting timeouts outbound from inside the container, works fine from the instance

### ISSUE THREAD

**Issue by rlewkowicz** [aws/amazon-vpc-cni-k8s#682]
From instance:
```
telnet my.splunk.url 9997
Trying 34.aws.ip.thing...
Connected to my.splunk.url.
Escape character is '^]'.
```

From pod:
```
telnet my.splunk.url 9997
Connection closed by foreign host
```

I can dig it, so dns resolves. There's also an event collector on 443 that I cannot reach, but I can curl things like google. There's no peering, but those splunk nodes are aws instances. 

I'm asking here only because I know that the cni has all sorts of weird integrations, but I don't know how to troubleshoot them.

**Comment by nithu0115** [comment]
@rlewkowicz would you please confirm the CNI version installed on the cluster by running below command ? If it returns 1.5.4, would you please patch/set downgrade the image to 1.5.3 ? 
‘kubectl describe daemonset aws-node --namespace kube-system | grep Image | cut -d "/" -f 2‘

**Comment by rlewkowicz** [comment]
@nithu0115 I'm on amazon-k8s-cni:v1.5.3

I also tried swtiching to traditional calico node and enabled nat'd output. Again, I could reach out to things like apk, but it just cant get to these specific aws instances but the host can.

**Comment by rlewkowicz** [comment]
I should add perhaps:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T23:49:20Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-eks-5ac0f1", GitCommit:"5ac0f1d9ab2c254ea2b0ce3534fd72932094c6e1", GitTreeState:"clean", BuildDate:"2019-08-20T22:39:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
```
I'm also using the eks amis.

**Comment by mogren** [comment]
@rlewkowicz Do you have some peered VPCs? Also, Calico might leave old rules around, usually listening with `tcpdump` on the L2 bridge between the pod network namespace and the host namespace can give a clue. Check the ip for the pod, then `ip route show table all` can give you the interface.

**Comment by rlewkowicz** [comment]
Wait a sec. It's arping to it's self?
```
IP ip-10-153-9-200.ec2.internal.56438 > ec2-34-199-173-126.compute-1.amazonaws.com.palace-6: Flags [S], seq 1143412828, win 26883, options [mss 8961,sackOK,TS val 248527207 ecr 0,nop,wscale 7], length 0
16:35:59.475096 IP ec2-34-199-173-126.compute-1.amazonaws.com.palace-6 > ip-10-153-9-200.ec2.internal.56438: Flags [S.], seq 4221508325, ack 1143412829, win 26847, options [mss 1460,sackOK,TS val 3158143771 ecr 248527207,nop,wscale 7], length 0
16:35:59.475112 IP ip-10-153-9-200.ec2.internal.56438 > ec2-34-199-173-126.compute-1.amazonaws.com.palace-6: Flags [.], ack 1, win 211, options [nop,nop,TS val 248527208 ecr 3158143771], length 0
16:36:04.138428 ARP, Request who-has ip-10-153-9-200.ec2.internal tell ip-10-153-9-230.ec2.internal, length 28
16:36:04.138578 ARP, Reply ip-10-153-9-200.ec2.internal is-at 92:1c:c8:75:46:4b (oui Unknown), length 28
```

**Comment by deliahu** [comment]
I am also seeing outbound timeouts from within the container on AWS CNI v1.5.4. I'm not sure if this is related since @rlewkowicz is on v1.5.3; I'd be happy to file a new issue if it's a separate bug.

Steps to reproduce:

1. Spin up an EKS cluster

```bash
cat <<EOF | eksctl create cluster -f -
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: test
  region: us-west-2
  version: "1.14"
nodeGroups:
  - name: ng-1
    instanceType: p2.xlarge
    minSize: 1
    maxSize: 1
    desiredCapacity: 1
    ami: auto
    iam:
      withAddonPolicies:
        autoScaler: true
EOF

eksctl utils write-kubeconfig --region=us-west-2 --name=test
```

2. Install AWS CNI v1.5.4:

```bash
kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.5.4/config/v1.5/aws-k8s-cni.yaml
```

3. Spin up pods and check their internet access (I had to repeat this a few times before reproducing):

```bash
NUM=22

# create pods
for ((i=1;i<=$NUM;i++)); do
  cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: a${i}
spec:
  containers:
  - name: busybox
    image: busybox
    args:
    - sleep
    - "1000000"
EOF
done

# <wait for pods to be ready>

# test outbound internet access
for ((i=1;i<=$NUM;i++)); do
  echo -n "a${i} "
  kubectl exec "a${i}" -- /bin/sh -c "if nc -zw1 google.com 443; then echo 'ok'; else echo 'NO INTERNET'; fi"
done

# delete pods
for ((i=1;i<=$NUM;i++)); do
  kubectl delete pod "a${i}" --grace-period=0
done
```

**Comment by 2tim** [comment]
Having similar issues with the latest eks and cni right after upgrading from 1.4

I noticed when running a maintenance pod and hitting an external api. I set the “EXTERNALSNAT” option to true. I’ve had similar problems off and on with this.

**Comment by 2tim** [comment]
@deliahu Thanks for that tests script. I was about to write something like that. I noticed about 10 percent failure rate with 1.5.4. I swapped out google.com with rds.amazonaws.com and had similar results which also included one DNS failure. It also seemed to take a while to run.
@mogren In my case we upgraded from 1.4.1 on a 1.14 cluster to 1.5.4. We have `AWS_VPC_K8S_CNI_EXTERNALSNAT=true` as we have a Virtual Gateway setup to our datacenter. It seems a bit more stable after rolling back to 1.4.1 but I went from a 10% failure rate with @deliahu script to about 2% after rolling the version back. Should I look at something on the hosts? It is starting to feel like something is maybe altered on the hosts iptables config. I'm going to run the test on a health cluster to see if I get the same or similar results.

**Comment by nithu0115** [comment]
@2tim - there is a known issue with 1.5.4 CNI version. For more details, check #641

**Comment by 2tim** [comment]
@nithu0115 thanks for the note. After further exploring the issue I believe our problem stemmed from another issue. When applying the update from 1.4.1 to 1.5.4 we failed to set `AWS_VPC_K8S_CNI_EXTERNALSNAT=true` in our config. Apparently this put our host into a bad state. Once we realized that was missing we tried reapplying with it set to true and still had the issues mentioned above. In the end we restarted the nodes from the console and without a redeploy of the CNI everything came back up fine and we are no longer seeing the issue. The toggling of the `AWS_VPC_K8S_CNI_EXTERNALSNAT` back and forth seems to have caused the issue. I don't know if this should be handled as a bug but I believe it could be reproduced by toggling the value a few times. I noticed that kube2iam could get into the same state under certain conditions as it alters iptable settings on the host as well. I hope this helps anyone else facing this issue.
